src/Apropos/Description.hs

```diff
@@ -695,23 +691,24 @@ infix 1 :<->:
 {- |
   Logical expressions for matching description types.
 -}
-type Formula :: Type -> Type
-data Formula attr
-  = Var (Attribute Int attr)
+type Formula attr = Formula' (Attribute Int attr)
+type Formula' :: Type -> Type
+data Formula' attr
+  = Var attr
   | Yes
   | No
-  | Not (Formula attr)
-  | Formula attr :&&: Formula attr
-  | Formula attr :||: Formula attr
-  | Formula attr :++: Formula attr
-  | Formula attr :->: Formula attr
-  | Formula attr :<->: Formula attr
-  | All (Vector (Formula attr))
-  | Some (Vector (Formula attr))
-  | None (Vector (Formula attr))
-  | ExactlyOne (Vector (Formula attr))
-  | AtMostOne (Vector (Formula attr))
-  deriving stock (Generic)
+  | Not (Formula' attr)
+  | Formula' attr :&&: Formula' attr
+  | Formula' attr :||: Formula' attr
+  | Formula' attr :++: Formula' attr
+  | Formula' attr :->: Formula' attr
+  | Formula' attr :<->: Formula' attr
+  | All (Vector (Formula' attr))
+  | Some (Vector (Formula' attr))
+  | None (Vector (Formula' attr))
+  | ExactlyOne (Vector (Formula' attr))
+  | AtMostOne (Vector (Formula' attr))
+  deriving stock (Generic,Functor,Eq,Ord)
```

Comment:
The style guide injuncts against type aliases, and this change pollutes the exported namespace with similar types whose differences are not immediately apparent. Since `mapFormula`/`fmap` is used exactly once right now, I am inclined to leave it the way it is.

Reply:
This is also failing CI for formatting.